client/js/lounge.js

```diff
@@ -646,7 +672,15 @@ $(function() {
 		}
 	});
 
-	socket.on("names", renderChannelUsers);
+	socket.on("names", function(data) {
+		renderChannelUsers(data);
+
+		// Condensed actions support
+		var messages = chat.find("#chan-" + data.id).find(".messages");
+		data.users.forEach(function(user) {
+			messages.find(".user[data-name='" + user.name + "']").removeClass("disconnected");
```

Comment:
We can remove this code for now and move this code to #759 or whatever PR it will be.